src/server/set_family.cc

```diff
@@ -1567,7 +1567,9 @@ vector<long> SetFamily::SetFieldsExpireTime(const OpArgs& op_args, uint32_t ttl_
     pv->InitRobj(OBJ_SET, kEncodingStrMap2, ss);
   }
 
-  return ExpireElements((StringSet*)pv->RObjPtr(), values, ttl_sec);
+  auto ss = static_cast<StringSet*>(pv->RObjPtr());
+  ss->set_time(MemberTimeSeconds(op_args.db_cntx.time_now_ms));
```

Comment:
should this call be in FieldExpireTime or in SetFieldsExpireTime?
From the logic of hset this call is done in FieldExpireTime flow

Reply:
yes I understand that. I wrote this comment because from the hset functionality I see that ss->set_time(MemberTimeSeconds is called from the flow of FieldExpireTime used in FIELDTTL

Reply:
I'm not sure why set_time is used in hset `FieldExpireTime`. Calculation for object expiration does not depend on object `time_now_` private member variable.

Reply:
I think that the implementation should be the same for hmap and sets so maybe there is a bug in hmap, lets see.
@romange can you review this?

Reply:
so your fix in `SetFamily::SetFieldsExpireTime` is correct - you need to update time_now_ so that the expiry time will be correct and Adi is correct- we should update `time_now_` on every opportunity we touch the data structure, also in `SetFamily::FieldExpireTime` and similarly we should do the same in hset_family.cc

Reply:
@romange @adiholden I have looked more closely in code, we are setting `time_now_` in set family. Time will be updated from `StringSetWrapper` constructor. This object is created in function `GetExpiry` in `set_family.cc`.